src/generators/pedersen_gens.rs

```diff
@@ -23,58 +23,238 @@ use crate::protocols::ristretto_point_protocol::RistrettoPointProtocol;
 /// * `h_base`: the `ristretto255` basepoint;
 /// * `g_base`: the result of `ristretto255` SHA3-512
 /// hash-to-group on input `B_bytes`.
-#[derive(Copy, Clone, Debug)]
+#[derive(Clone, Debug)]
 pub struct PedersenGens {
     /// Base for the committed value
     pub h_base: RistrettoPoint,
-    /// Base for the blinding factor
-    pub g_base: RistrettoPoint,
+    /// Base for the blinding factor vector
+    pub g_base_vec: Vec<RistrettoPoint>,
     /// Compressed base for the committed value
     pub h_base_compressed: CompressedRistretto,
-    /// Compressed base for the blinding factor
-    pub g_base_compressed: CompressedRistretto,
+    /// Compressed base for the blinding factor vector
+    pub g_base_compressed_vec: Vec<CompressedRistretto>,
+    /// Blinding factor extension degree
+    pub extension_degree: ExtensionDegree,
+}
+
+/// The extension degree for extended commitments
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum ExtensionDegree {
+    /// Default Pedersen commitment
+    ZERO = 1,
+    /// Pedersen commitment extended with one degree
+    ONE = 2,
+    /// Pedersen commitment extended with two degrees
+    TWO = 3,
 }
 
 lazy_static! {
-    static ref RISTRETTO_BASEPOINT_POINT_BLINDING: RistrettoPoint =
+    static ref RISTRETTO_BASEPOINT_POINT_BLINDING_1: RistrettoPoint =
         RistrettoPoint::hash_from_bytes_sha3_512(RISTRETTO_BASEPOINT_COMPRESSED.as_bytes());
+    static ref RISTRETTO_BASEPOINT_POINT_BLINDING_2: RistrettoPoint = RistrettoPoint::hash_from_bytes_sha3_512(
+        RistrettoPoint::hash_from_bytes_sha3_512(RISTRETTO_BASEPOINT_COMPRESSED.as_bytes())
+            .compress()
+            .as_bytes()
+    );
+    static ref RISTRETTO_BASEPOINT_POINT_BLINDING_3: RistrettoPoint = RistrettoPoint::hash_from_bytes_sha3_512(
+        RistrettoPoint::hash_from_bytes_sha3_512(
+            RistrettoPoint::hash_from_bytes_sha3_512(RISTRETTO_BASEPOINT_COMPRESSED.as_bytes())
+                .compress()
+                .as_bytes()
+        )
+        .compress()
+        .as_bytes()
+    );
+    static ref RISTRETTO_BASEPOINT_COMPRESSED_BLINDING_2: CompressedRistretto =
+        (*RISTRETTO_BASEPOINT_POINT_BLINDING_2).compress();
+    static ref RISTRETTO_BASEPOINT_COMPRESSED_BLINDING_3: CompressedRistretto =
+        (*RISTRETTO_BASEPOINT_POINT_BLINDING_3).compress();
 }
```

Comment:
Recommend simply hashing a unique indexed string to generate these points, rather than iterated hashing. This is less likely to result in unintended collisions or reuse elsewhere.